src/services/Evernode.js

```diff
@@ -119,4 +121,71 @@ const onLedger = async (callback) => {
             moment: moment
         })
     });
+}
+
+const testnetFaucet = async() => {
+    const generatedAccount = await generateAndFundFaucetAccount();
+    return generatedAccount;
+}
+
+const generateAndFundFaucetAccount = async() => {
+    const xrplServerURL = "wss://hooks-testnet-v2.xrpl-labs.com";
+    const xrplClient = new xrpl.Client(xrplServerURL);
+    await xrplClient.connect();
+    try{
+        // Generating faucet account
+        const new_wallet = xrpl.Wallet.generate();
+
+        // Polling happens every 5 seconds
+        return new Promise(async(resolve, reject) => {
+            await fetch(`https://hooks-testnet-v2.xrpl-labs.com/newcreds?account=${new_wallet.address}`, {
+                method: 'POST',
+                mode: 'no-cors',
+                headers: {
+                    "Content-Type": "application/json"
+                }
+            }).then(async()=> {
+                const hostClient = new evernode.HostClient(new_wallet.address, new_wallet.seed);
+                await hostClient.connect();
+
+                console.log("Requesting beta EVRs...");
+            
+                await hostClient.xrplAcc.setTrustLine(FaucetAccount.EVR, hostClient.config.evrIssuerAddress, "99999999999999");
+
+                await hostClient.xrplAcc.makePayment(hostClient.config.foundationAddress,
+                    evernode.XrplConstants.MIN_XRP_AMOUNT,
+                    evernode.XrplConstants.XRP,
+                    null,
+                    [{ type: 'giftBetaEvr', format: '', data: '' }]);
+
+                // Keep watching our EVR balance.
+                let attempts = 0;
+                while (attempts >= 0) {
+                    await new Promise(resolve => setTimeout(resolve, 1000));
```

Comment:
Wouldn't this be ambiguous since the parent resolve function is also in the same name